crates/evm-coder/src/abi.rs

```diff
@@ -365,12 +360,16 @@ macro_rules! impl_tuples {
 			($($ident,)+): SolidityTypeName,
 		{
 			fn abi_read(&mut self) -> Result<($($ident,)+)> {
-				let size = if <($($ident,)+)>::is_simple() { Some(0 $(+aligned_size(sp_std::mem::size_of::<$ident>()))+) } else { None };
+				let size = if <($($ident,)+)>::is_simple() { Some(<Self as AbiRead<($($ident,)+)>>::size()) } else { None };
 				let mut subresult = self.subresult(size)?;
 				Ok((
 					$(<Self as AbiRead<$ident>>::abi_read(&mut subresult)?,)+
 				))
 			}
+
+			fn size() -> usize {
+				0 $(+ {let _ : $ident; ABI_ALIGNMENT})+
```

Comment:
```suggestion
				0 $(+ $ident::size())+
```
Для поддержки tuple внутри tuple

Reply:
Только так не получиться, т.к. у нас простые типы не имплементируют трэйт AbiRead

Reply:
Это какие?

Reply:
Там есть макрос impl_abi_readable, который как раз это реализует